src/main/java/org/gridsuite/odre/server/OdreController.java

```diff
@@ -42,4 +55,49 @@ public void pushSubstations() {
     public void pushLines() {
         odreService.pushLines();
     }
+
+    @PostMapping(value = "/lines/upload", consumes = "multipart/form-data")
+    @Operation(summary = "Get lines coordinates from csv files and send them to geo data service")
+    @ApiResponses (value = {@ApiResponse(responseCode = "200", description = "the list of lines was updated"),
+            @ApiResponse(responseCode = "500", description = "fail to upload file(s)"),
+            @ApiResponse(responseCode = "400", description = "csv file with invalid name or no file attached"),
+    })
+    public ResponseEntity<FileUploadResponse> pushLinesFromCsv(@RequestParam("files") MultipartFile[] files) {
+        ResponseEntity<FileUploadResponse> response;
+        try {
+            Map<String, MultipartFile> fileMap = new HashMap<>();
+            //check if all files are present
+            if (Arrays.stream(files).distinct().filter(file -> GeographicDataParser.hasCSVFormat(file) && FileNameEnum.checkIfValueExist(file.getOriginalFilename())).count() < 3) {
```

Comment:
I would have put those checks into odreService.pushLinesFromCsv scope to avoid beint dependent her of GeographicDataParser but well ok.

Reply:
ill try to check from the begining to avoid entring into service and make a return unstead of void , if it's really disturbing just confirm it to me and ill do it.